### PR TITLE
adding IID and NonIID dataloading parsers for cifar10, cifar100, calt…

### PR DESCRIPTION
**Pull request by sahiltyagi4**
…ech101 and caltech256. partitions iid data evenly among clients; for non-iid data, partitions by splitting data on a per-label basis and distributing evenly among clients (under #55)

### CODE REVIEW

**Comment by sahiltyagi4** [comment]
@andreicozma1 added DataLoaderIID and DataLoaderNonIID classes in the data module. each has functions CIFAR10, CIFAR100, CalTech101 and CalTech256. 
- IID data is iid partitioned evenly among clients: For e.g., dataset of 100 samples are split among 4 clients in iid manner as 25 samples per client.
- Non-IID data is split evenly among clients on a per-label basis. E.g., CIFAR100 with 10 clients is split by allocating samples with 10 unique labels to each client.
- Currently, both classes take arguments like client_id, total_clients, total_labels, train batch-size, test batch-size, etc. Does this setup work or do you prefer to move the class arguments down to function-wise arguments. Or do you prefer one function per-class (i.e., separate classes for iid cifar10, iid cifar100, non-iid cifar10, etc.)
